build.sh fails on Linux due to a unitialised Git submodule

### Issue
- `build.sh` fails on Linux as libmf which is included as a Git submodule is not available unless submodules have been initialized.

### Logs
```
  + cmake /home/tpalohei/machinelearning-foo/src/Native -G 'Unix Makefiles' -DCMAKE_BUILD_TYPE=Debug -DMKL_LIB_PATH=/home/tpalohei/machinelearning-foo/packages/mlnetmkldeps/0.0.0.7/runtimes/linux-x64/native -DMKL_LIB_RPATH=../../../../../microsoft.ml.mkl.redist//runtimes/linux-x64/native -DVERSION_FILE_PATH:STRING=/home/tpalohei/machinelearning-foo/src/Native/../../bin/obj/version.c
  -- The C compiler identification is Clang 7.0.0
  -- The CXX compiler identification is Clang 7.0.0
  -- Check for working C compiler: /usr/lib/llvm/7/bin/clang
  -- Check for working C compiler: /usr/lib/llvm/7/bin/clang -- works
  -- Detecting C compiler ABI info
  -- Detecting C compiler ABI info - done
  -- Detecting C compile features
  -- Detecting C compile features - done
  -- Check for working CXX compiler: /usr/lib/llvm/7/bin/clang++
  -- Check for working CXX compiler: /usr/lib/llvm/7/bin/clang++ -- works
  -- Detecting CXX compiler ABI info
  -- Detecting CXX compiler ABI info - done
  -- Detecting CXX compile features
  -- Detecting CXX compile features - done
  -- Configuring done
  CMake Error at MatrixFactorizationNative/CMakeLists.txt:14 (add_library):
    Cannot find source file:
  
      libmf/mf.cpp
  
    Tried extensions .c .C .c++ .cc .cpp .cxx .cu .m .M .mm .h .hh .h++ .hm
    .hpp .hxx .in .txx
  
  
  -- Build files have been written to: /home/tpalohei/machinelearning-foo/bin/obj/x64.Debug/Native
  CMake Error at MatrixFactorizationNative/CMakeLists.txt:14 (add_library):
    No SOURCES given to target: MatrixFactorizationNative
```

--- a/docs/building/unix-instructions.md
+++ b/docs/building/unix-instructions.md
@@ -3,7 +3,7 @@ Building ML.NET on Linux and macOS
 ## Building
 
 1. Install the prerequisites ([Linux](#user-content-linux), [macOS](#user-content-macos))
-2. Clone the machine learning repo `git clone https://github.com/dotnet/machinelearning.git`
+2. Clone the machine learning repo `git clone --recursive https://github.com/dotnet/machinelearning.git`
 3. Navigate to the `machinelearning` directory
 4. Run the build script `./build.sh`
 

